sup3r/bias/bias_correct_means.py

```diff
@@ -0,0 +1,949 @@
+
+        Parameters
+        ----------
+        data : xarray.Dataset
+            xarray dataset object for a year of ERA data. Must include u/v
+            components for the given height. e.g. u_{height}m, v_{height}m.
+        height : int
+            Height to compute windspeed for.
+        """
+        return np.hypot(
+            data[f"u_{height}m"].values, data[f"v_{height}m"].values
+        )
+
+    def get_month_mean(self, data, height, month):
+        """Get windspeed_{height}m mean for the given month.
+
+        Parameters
+        ----------
+        data : xarray.Dataset
+            xarray dataset object for a year of ERA data. Must include u/v
+            components for the given height. e.g. u_{height}m, v_{height}m.
+        height : int
+            Height to compute windspeed for.
+        month : int
+            Index of month to get mean for. e.g. 1 = Jan, 2 = Feb, etc.
+
+        Returns
+        -------
+        out : np.ndarray
+            Array of time averaged windspeed data for the given month.
+
+        """
+        mask = pd.to_datetime(data["time"]).month == month
+        ws = self.get_windspeed(data, height)[mask]
+        return ws.mean(axis=0)
+
+    def get_all_means(self, height):
+        """Get monthly means for all months across all given years for the
+        given height.
+
+        Parameters
+        ----------
+        height : int
+            Height to compute windspeed for.
+
+        Returns
+        -------
+        means : dict
+            Dictionary of windspeed_{height}m means for each month
+        """
+        feature = self.features[self.heights.index(height)]
+        means = {i: [] for i in range(1, 13)}
+        for i, year in enumerate(self.years):
+            logger.info(f"Getting means for year={year}, feature={feature}.")
+            data = xr.open_dataset(self.input_files[i])
+            for m in range(1, 13):
+                means[m].append(self.get_month_mean(data, height, month=m))
+        means = {m: np.dstack(arr).mean(axis=-1) for m, arr in means.items()}
+        return means
+
+    def write_csv(self, out, out_file):
+        """Write monthly means to a csv file.
+
+        Parameters
+        ----------
+        out : dict
+            Dictionary of windspeed_{height}m means for each month
+        out_file : str
+            Name of csv output file.
+        """
+        logger.info(f"Writing means to {out_file}.")
+        out = {
+            f"{str(calendar.month_name[m])[:3]}_mean": v.flatten()
+            for m, v in out.items()
+        }
+        df = pd.DataFrame.from_dict(out)
+        df["latitude"] = self.lats.flatten()
+        df["longitude"] = self.lons.flatten()
+        df["gid"] = np.arange(len(df["latitude"]))
+        df.to_csv(out_file)
+        logger.info(f"Finished writing means for {out_file}.")
+
+    @classmethod
+    def run(cls, era_pattern, years, features, out_pattern):
```

Comment:
The ERA netcdf files should be able to be handled by the standard DataHandlerNC classes and the BiasCalc class. Why didnt you just convert vortex data to a standard format and then use the BiasCalc class from there? I think the mods to allow the BiasCalc module to use monthly data would be really simple